### ISSUE TITLE
There is no information about Remaining requests.

### ISSUE THREAD

**Issue by LinaKorn** [Azure/azure-sdk-for-java#1781]
In the Throttling Resource Manager requests Azure have an information about Remaining requests.  Each request includes values for the number of remaining read and write requests. But information from request header is not processing in SDK. Is it a correct implementation? How we can retrieve the number of remaining requests via SDK?

We are using Azure SDK 0.9.7 (Does current problem solved in the version 1.0.0)

**Comment by hovsepm** [comment]
@LinaKorn could you post http response example that contains remaining reads? I'm pretty sure it is not available neither in 0.9.7 nor in 1.0.0. But we may consider this as a bugfix in the next releases.

**Comment by LinaKorn** [comment]
`{
  "statusCode": 200,
  "header": {
    "cache-control": "no-cache",
    "pragma": "no-cache",
    "content-type": "application/json; charset=utf-8",
    "expires": "-1",
    "x-ms-ratelimit-remaining-subscription-reads": "14998",`

All remaining requests that can be in header you may find in [Throttling Resource Manager requests](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-request-limits)

**Comment by tmonning** [comment]
This still seems to be unavailable in 1.0.0 plus I often get the following error (although I think I haven't had many calls):
`com.microsoft.azure.CloudException: Status code 429, {"error":{"code":"TooManyRequests","message":"The request is being throttled."}}: The request is being throttled.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at com.microsoft.rest.ServiceResponseBuilder.build(ServiceResponseBuilder.java:120)
	at com.microsoft.azure.AzureResponseBuilder.build(AzureResponseBuilder.java:56)
	at com.microsoft.azure.management.storage.implementation.StorageAccountsInner.listByResourceGroupDelegate(StorageAccountsInner.java:787)
	at com.microsoft.azure.management.storage.implementation.StorageAccountsInner.access$600(StorageAccountsInner.java:46)
	at com.microsoft.azure.management.storage.implementation.StorageAccountsInner$26.call(StorageAccountsInner.java:773)
	at com.microsoft.azure.management.storage.implementation.StorageAccountsInner$26.call(StorageAccountsInner.java:769)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.onNext(OnSubscribeMap.java:69)
	at retrofit2.adapter.rxjava.RxJavaCallAdapterFactory$RequestArbiter.request(RxJavaCallAdapterFactory.java:173)
	at rx.Subscriber.setProducer(Subscriber.java:211)
	at rx.internal.operators.OnSubscribeMap$MapSubscriber.setProducer(OnSubscribeMap.java:102)
	at retrofit2.adapter.rxjava.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:152)
	at retrofit2.adapter.rxjava.RxJavaCallAdapterFactory$CallOnSubscribe.call(RxJavaCallAdapterFactory.java:138)
	at rx.Observable.unsafeSubscribe(Observable.java:10142)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:48)
	at rx.internal.operators.OnSubscribeMap.call(OnSubscribeMap.java:33)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:48)
	at rx.internal.operators.OnSubscribeLift.call(OnSubscribeLift.java:30)
	at rx.Observable.subscribe(Observable.java:10238)
	at rx.Observable.subscribe(Observable.java:10205)
	at rx.observables.BlockingObservable.blockForSingle(BlockingObservable.java:444)
	at rx.observables.BlockingObservable.single(BlockingObservable.java:341)
	at com.microsoft.azure.management.storage.implementation.StorageAccountsInner.listByResourceGroup(StorageAccountsInner.java:714)
	at com.microsoft.azure.management.resources.fluentcore.arm.collection.implementation.TopLevelModifiableResourcesImpl.listByResourceGroup(TopLevelModifiableResourcesImpl.java:121)`

**Comment by LinaKorn** [comment]
@hovsepm Could you tell me when it will be fixed?

**Comment by tmonning** [comment]
For us we found a workaround by registering an interceptor that performs the throttling (I don't know if this is a good approach) but having something like this being configurable directly in the SDK would be nice:

`Azure.configure().withInterceptor(new AzureThrottlingInterceptor())`

```
public class AzureThrottlingInterceptor implements Interceptor
{
	private static Logger logger = Logger.getLogger(AzureThrottlingInterceptor.class.getName());

	@Override
	public Response intercept(Interceptor.Chain chain) throws IOException
	{
		Request request = chain.request();

		Response response = chain.proceed(request);

		if (response.code() == 429)
		{
			logger.warn("Throttling occurred");
			String retryValue = response.header("Retry-After");
			if (retryValue != null && !retryValue.isEmpty())
			{
				response.close();
				int retryValueInt = Integer.valueOf(retryValue);
				logger.warn("Throttling for " + retryValueInt + "s");
				Thread.sleep(retryValueInt * 1000);
				return intercept(chain);
			}
		}

		return response;
	}
}
```

**Comment by jianghaolu** [comment]
An interceptor to handle 429s will be added in 1.2.0.

**Comment by martinsawicki** [comment]
released in v1.2
